lib/modules/manager/vendir/artifacts.ts

```diff
@@ -0,0 +1,123 @@
+import is from '@sindresorhus/is';
+import { TEMPORARY_ERROR } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+import {
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+export async function updateArtifacts({
+  packageFileName,
+  updatedDeps,
+  newPackageFileContent,
+  config,
+}: UpdateArtifact): Promise<UpdateArtifactsResult[] | null> {
+  logger.debug(`vendir.updateArtifacts(${packageFileName})`);
+
+  const isLockFileMaintenance = config.updateType === 'lockFileMaintenance';
+  const isUpdateOptionAddVendirArchives = config.postUpdateOptions?.includes(
+    'vendirUpdateSubChartArchives',
+  );
+
+  if (
+    !isLockFileMaintenance &&
+    (updatedDeps === undefined || updatedDeps.length < 1)
+  ) {
+    logger.debug('No updated vendir deps - returning null');
+    return null;
+  }
+
+  const lockFileName = getSiblingFileName(packageFileName, 'vendir.lock.yml');
+  const existingLockFileContent = await readLocalFile(lockFileName, 'utf8');
+  if (!existingLockFileContent && !isUpdateOptionAddVendirArchives) {
+    logger.debug('No vendir.lock.yml found');
+    return null;
+  }
+  try {
+    await writeLocalFile(packageFileName, newPackageFileContent);
+    logger.debug('Updating Vendir artifacts');
+    const execOptions: ExecOptions = {
+      cwdFile: packageFileName,
+      toolConstraints: [
+        { toolName: 'vendir', constraint: config.constraints?.vendir },
+        { toolName: 'helm', constraint: config.constraints?.helm },
+      ],
+    };
+
+    await exec(`vendir sync`, execOptions);
+
+    logger.debug('Returning updated Vendir artifacts');
+
+    const fileChanges: UpdateArtifactsResult[] = [];
+
+    if (is.truthy(existingLockFileContent)) {
+      const newVendirLockContent = await readLocalFile(lockFileName, 'utf8');
+      const isLockFileChanged =
+        existingLockFileContent !== newVendirLockContent;
+      if (isLockFileChanged) {
+        fileChanges.push({
+          file: {
+            type: 'addition',
+            path: lockFileName,
+            contents: newVendirLockContent,
+          },
+        });
+      } else {
+        logger.debug('vendir.lock.yml is unchanged');
+      }
+    }
+
+    // add modified vendir archives to artifacts
+    if (is.truthy(isUpdateOptionAddVendirArchives)) {
+      logger.debug("Adding Sync'd files to git");
+      // Files must be in the vendor path to get added
+      const vendorDir = getSiblingFileName(packageFileName, '.');
```

Comment:
```suggestion
      const vendorDir = getParentDir(packageFileName);
```
you want the parent folder of filename?

Reply:
Yes, I wanted everything under the parent dir. Updated to use getParentDir